### ISSUE TITLE
Runner registration check instantly restarting pods

### ISSUE THREAD

**Issue by BrendanGalloway** [summerwind/actions-runner-controller#300]
All runner pods are instantly terminating, likely due to the changes in #297 

The controller log states:
```Runner failed to register itself to GitHub in timely manner. Recreating the pod to see if it resolves the issue. CAUTION: If you see this a lot, you should investigate the root cause. See https://github.com/summerwind/actions-runner-controller/issues/288	"runner": "github-runner/flanksource-ci-runner-deployment-p6g48-pd6qg", "podCreationTimestamp": "2021-02-09 07:56:58 +0000 UTC", "currentTime": "2021-02-09T07:56:58.547Z", "configuredRegistrationTimeout": "10m0s"}```
however it seems like there is some bug in this logic as the pod is not waiting 10 min to register.

**Comment by jwnx** [comment]
A few minutes ago I started experiencing this in my testing environment. Haven't found a workaround yet.

**Comment by iautom8things** [comment]
@jwnx I solved this by pinning the image tag to `v1.17.0`

values for helm release:

```yaml
# ...
image:
  dindSidecarRepositoryAndTag: docker:dind
  pullPolicy: IfNotPresent
  repository: summerwind/actions-runner-controller
  tag: v0.17.0
```

**Comment by mumoshu** [comment]
Sorry everyone this seems like a bug in the main branch as @BrendanGalloway has pointed out.

Was it your production environment? If so, I got to think if we'd better update the `latest` docker image tag only on new releases, and use another tag for the image that is updated on every commit to the main branch.
